.githooks/pre-commit

```diff
@@ -1,22 +1,38 @@
 #!/bin/bash
 cd "$(git rev-parse --show-toplevel)"
+# Pylint Path is Micky's local path, feel free to change in PR 
+PYLINT = "c:\users\micky\appdata\local\programs\python\python310\lib\site-packages" 
 ESLINT="client/node_modules/.bin/eslint"
 pwd
 
-staged=`git diff --name-only --cached --diff-filter=A --diff-filter=M | cat`
-pylint="c:\users\micky\appdata\local\programs\python\python310\lib\site-packages  --rcfile .pylintrc --load-plugins pylint_django --load-plugins pylint_quotes --reports=no"
+if [[ ! -x "$PYLINT" ]]; then
+	  printf "\t\033[41mPlease install Pylint\033[0m (pip install pylint)\n"
+	    exit 1
+    fi
+
+    STAGED_FILES=($(git diff --cached --name-only --diff-filter=ACM | grep ".py\{0,1\}$"))
+
+    echo "Checking ${#STAGED_FILES[@]} staged files for Pylint Errors..."
+
+    if [[ "$STAGED_FILES" = "" ]]; then
+	      exit 0
+      fi
+
+      $PYLINT "${STAGED_FILES[@]}" --fix
```

Comment:
I think in this line you'll also have to add in the pylint flags like `--rcfile \path\to\.pylintrc --load-plugins pylint_django`

Apparently the pylint_django extension is important to install for our implementation or else a bunch of excessive error messages will be thrown

Reply:
Also I checked and pylint does not come with a `--fix` flag.